lib/logger.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @file Abstracted logging properties and methods.
+ *
+ * If for any reason, they need a different logging tool or need to be extended,
+ * the changes can be made here.
+ */
+var merge = require('lodash.merge');
+var util = require('util');
+var winston = require('winston');
+
+/**
+ * Abstracts fully-fleshed logging module to the logger property of this module.
+ *
+ * If logging with metadata, or requiring winston.log's more advanced features,
+ * use exports.logger.log() instead of exports.log().
+ * @example
+ *   exports.logger(options).log('info', 'Test Log Message', { anything: 'This is metadata' });
+ *
+ * @param {Object} options - Configuration options.
+ * @return {Object} An instance of Winston.
+ */
+exports.logger = function (options) {
+  'use strict';
+
+  options = getOptions(options);
+
+  var exitOnError = options.logger.exitOnError;
+  var level = options.logger.level;
+  var logFile = options.logger.logFile;
+  var silent = options.logger.silent;
+  var transports = [
+    new winston.transports.Console({
```

Comment:
@scottnath 
cc: @rebmullin @sergesemashko @conortm @gvorbeck @beynya @nikkiana
Please note that a Winston transports object has to be instantiated EVERY time a logger method is fired. And the only reason we need this is to pass a configuration param. Because this app is extremely procedural, and not object-oriented, I wasn't able to devise an easy way to keep instantiated objects as properties of a global runtime object.

Another procedure that should be reworked is the getOptions() method I copied and pasted to the bottom of this file. Again, why don't we set the default options once and only once? Then, we won't need to repeat ourselves for every module on this project.

I think I either need to know what I'm missing if I am missing a simpler, less repetitive ways of doing things, or else we should consider rewriting things to be more object-oriented, simpler if possible, and definitely less brittle. Since I'd rather make suggestion for improvement than just criticize, I propose that I find a way to establish default configs once and only once.